spec.bs

```diff
@@ -597,41 +614,39 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
+    [=auction config/per buyer signals=][|buyer|] [=map/exists=]:
+    1. Set |perBuyerSignals| to |auctionConfig|'s [=auction config/per buyer signals=][|buyer|].
   1. [=map/For each=] |signalsUrl| -> |perSignalsUrlGenerator| of |perBuyerGenerator|:
     1. Let |keys| be a new [=ordered set=].
     1. Let |igNames| be a new [=ordered set=].
     1. [=map/For each=] joiningOrigin -> |groups| of |perSignalsUrlGenerator|:
       1. [=list/For each=] |ig| of |groups|:
         1. [=set/Append=] |ig|'s [=interest group/trusted bidding signals keys=] to |keys|.
         1. [=set/Append=] |ig|'s [=interest group/name=] to |igNames|.
-    1. Let |fullSignalsUrl| be the result of [=building trusted bidding signals url=] with
+    1. Let |biddingSignalsUrl| be the result of [=building trusted bidding signals url=] with
       |signalsUrl|, |keys|, |igNames|, |buyerExperimentGroupId|.
-    1. Let |trustedBiddingSignals| be the result of [=fetching resource=] with |fullSignalsUrl|,
-      "application/json", and null.
+    1. Let « |trustedBiddingSignals|, |dataVersion| » be the result of [=fetching trusted signals=]
+      with |biddingSignalsUrl| and true.
     1. [=map/For each=] joiningOrigin -> |groups| of |perSignalsUrlGenerator|:
       1. [=list/For each=] |ig| of |groups|:
         1. TODO: Let |interestGroup| be ... from |ig| ... minus priority and prioritySignalsOverrides and any browser-defined pieces
-        1. Set |browserSignals|'s "joinCount" to the sum of |ig|'s
+        1. [=map/Set=] |browserSignals|["`joinCount`"] to the sum of |ig|'s
            [=interest group/join counts=] for all days within the last 30 days.
-        1. Set |browserSignals|'s "bidCount" to the sum of |ig|'s
+        1. [=map/Set=] |browserSignals|["`bidCount`"] to the sum of |ig|'s
            [=interest group/bid counts=] for all days within the last 30 days.
-        1. Set |browserSignals|'s "prevWins" to a [=list=] containing
+        1. [=map/Set=] |browserSignals|["`prevWins`"] to a [=list=] containing
            [=tuple=]s of the time and the corresponding winning
            [=interest group ad=] from |ig|'s [=interest group/previous wins=]
            field with a data within the last 30 days. The time field is
            specified in seconds relative to the start of the auction.
         1. TODO: Set |browserSignals|'s "wasmHelper" ...need to fetch and prepare WebAssembly.Module object based on |ig|'s [=interest group/bidding wasm helper url=]
-        1. TODO: Set |browserSignals|'s "dataVersion" ...to Data-Version value from the trusted bidding signals server's response(s)
-        1. Let |biddingScript| be the result of [=fetching resource=] with |ig|'s
-          [=interest group/bidding url=], "text/javascript", and null. If |biddingScript| is an error,
-          [=iteration/continue=].
-        1. Let |generatedBidResult| be the result of [=evaluating a bidding script=] with

```

Comment:
There are still some usages of the name |generatedBidResult| below.

Also, I recall that we were considering modifying [=evaluate a bidding script=] so that it can take the expected WebIDL type as a parameter, and do validation? I think @jyasskin mentioned this? Also tagging @domfarolino.

Reply:
> There are still some usages of the name |generatedBidResult| below.

I think I fixed this by putting it back to `generatedBidResult` here.

> Also, I recall that we were considering modifying [=evaluate a bidding script=] so that it can take the expected WebIDL type as a parameter, and do validation? I think @jyasskin mentioned this? Also tagging @domfarolino.

It looks like the conversion and validation happen below on the next eight lines.

Reply:
> It looks like the conversion and validation happen below on the next eight lines.

Right, but I think we wanted to put some of that logic in [=evaluate a bidding script=] instead of after every invocation of that algorithm...anyways, it doesn't have to happen in this PR.

Reply:
[=evaluate a bidding script=] is only invoked once, so it wouldn't save lines of spec, and the three consecutive [=evaluate a ...=] definitions have a nice consistency to them that might be disrupted if one of them did return value conversion.  I don't feel strongly.

Reply:
Oops, I meant [=evaluate a script=], not [=evaluate a bidding script=].

Reply:
Hmm, lets consider this for another PR.